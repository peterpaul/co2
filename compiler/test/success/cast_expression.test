["stdio.h"] int printf(char * format, ...);

interface Baz
{
}

class ObjectA <Baz>
{
	int foo;
}

class ObjectB : ObjectA
{
	int bar;
}

int main(int argc, char ** argv)
{
	unsigned i = 5;

	int test_float = (int) 5.6;

	printf("test_float = %d\n", test_float);

	ObjectA objb = (ObjectA) new ObjectB();

	ObjectA obja = new ObjectA();

	Baz bar = (Baz) obja;

	Baz baz = (Baz) objb;
}