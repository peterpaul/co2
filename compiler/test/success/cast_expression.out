test_float = 5
